main.py

```diff
@@ -45,77 +49,100 @@ def get_eurojackpot_results():
         f"https://www.veikkaus.fi/api/draw-results/v1/games/EJACKPOT/draws/by-week/{year}-W{week}").json()
 
     if len(r) == 0:
-        return None, None, None
-
-    results = r[-1]["results"]
-
-    primary_result = results[0]["primary"]
-    secondary_result = results[0]["secondary"]
-
-    prize_tiers = r[-1]["prizeTiers"]
+        return []
 
-    return primary_result, secondary_result, prize_tiers
+    return [EuroJackpot(e) for e in r]
 
 
-def generate_discord_msg():
-    primary_result, secondary_result, prize_tiers = get_eurojackpot_results()
-    if not primary_result:
-        return "Tuloksia ei saatu Veikkaukselta :("
+def fetch_winnings(
+        eurojackpot: EuroJackpot,
+        guesses_primary: List[str],
+        guesses_secondary: List[str]) -> Tuple[int, int, int, int]:
+    primary_results = eurojackpot.results[0].primary
+    secondary_results = eurojackpot.results[0].secondary
 
     primary_hits = 0
-    for number in primary_numbers:
-        if number in primary_result:
+    for number in primary_results:
+        if number in guesses_primary:
             primary_hits += 1
 
     secondary_hits = 0
-    for number in secondary_numbers:
-        if number in secondary_result:
+    for number in secondary_results:
+        if number in guesses_secondary:
             secondary_hits += 1
 
-    result = f"{primary_hits}+{secondary_hits} oikein"
+    total_hits = f"{primary_hits}+{secondary_hits} oikein"
 
     money_won = 0
-
-    for tier in prize_tiers:
-        if tier["name"] == result:
-            money_won = int(tier["shareAmount"])
+    for prize_tier in eurojackpot.prize_tiers:
+        if prize_tier.name == total_hits:
+            money_won = prize_tier.share_amount
+            break
 
     investment_value_old = get_investment_value()
-    investment_value = str(investment_value_old - 200 + money_won)
+    investment_value_new = investment_value_old - 200 + money_won
+
+    set_investment_value(investment_value_new)
```

Comment:
`investment_value_new` should be string in this case, or maybe it would be better to be converted value to string at the set and get functions

Something like 
```
def get_investment_value() -> int:
    result = ssm.get_parameter(Name=parameter_store_variable_name)
    return int(result["Parameter"]["Value"])


def set_investment_value(value: int):
    ssm.put_parameter(Name=parameter_store_variable_name,
                      Overwrite=True, Value=str(value))
```

Reply:
Great point I completely overlooked since I have never used AWS nor `boto3` library. Both accessor functions are now exactly asin your comment.